### PR TITLE
add redis process time to getStatus

### LINKED ISSUE #498: Time redis seperate from postgres
Process each promise with their own .then to roughly measure the latency of postgres vs redis

### PR DESCRIPTION
**Pull request by mchangrh**
closes #498

### CODE REVIEW

**Comment by mchangrh** [comment]
Added a timeout clause at 5 seconds to help better debug instead of taking 10+ minutes - this will allow us to differentiate between postgres and redis errors at a glance

File: src/routes/getStatus.ts [hunk 1]
```diff
@@ -3,25 +3,40 @@ import { Logger } from "../utils/logger";
...
 
 export async function getStatus(req: Request, res: Response): Promise<Response> {
     const startTime = Date.now();
     let value = req.params.value as string[] | string;
     value = Array.isArray(value) ? value[0] : value;
+    let processTime, redisProcessTime = -1;
+    const timeoutPromise = async (): Promise<string> => {
+        await setTimeout(5000);
+        return "timeout";
+    };
```
Comment (ajayyy) [inline src/routes/getStatus.ts:16]: Can you use promiseTimeout from util?

**Comment by mchangrh** [comment]
not able to test the timeouts but it should catch most cases

File: src/routes/getStatus.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export async function getStatus(req: Request, res: Response): Promise<Response>
             })
             .catch(e => {
                 Logger.error(`status: SQL query timed out: ${e}`);
+                return -1;
```
Comment (ajayyy) [inline src/routes/getStatus.ts:21]: This does nothing, it returns inside the callback

Comment (ajayyy) [reply src/routes/getStatus.ts:21]: oh nvm, does this keep passing to the fulfilled promise?

**Pull request merged**